code/modules/clothing/clothing.dm

```diff
@@ -217,6 +217,27 @@
 		for(var/A in armor)
 			armor[A] -= rand(armor[A]/3, armor[A])
 
+// Redefines the attack proc so that when we attack somebody on fire on help intent with a piece of clothing, we have a small chance of putting out the fire.
+/obj/item/clothing/attack(mob/living/M, mob/living/user, def_zone, var/originator = null)
+	if(iscarbon(user)) // First, we check if it's a a carbon.
+		if(user.a_intent == I_HELP) // Then we check if he wants to help.
+			if(istype(src, /obj/item/clothing/under) || (istype(src, /obj/item/clothing/suit) && !istype(src, /obj/item/clothing/suit/space))) // Then we check if the item is either a jumpsuit or suit-but-not-a-spacesut
+				if(ishuman(M) && M.on_fire) // If he is on fire
+					var/mob/living/carbon/human/target = M
+					// Good old polymorphism. We are allowed to do that because we checked if M was a human before.
+					// This allows us to check for their species
+					if(isplasmaman(target)) // Cannot put out plasmamen, else they could just go arround with a jumpsuit and not need a space suit.
+						visible_message("<span class='warning'>\The [user] attempts to put out the fire on \the [target], but plasmafires are too hot. It is no use.</span>")
+						return
+					else
+						visible_message("<span class='warning'>\The [user] attempts to put out the fire on \the [target] with \the [src].</span>")
+						if(prob(15) || (istype(src, /obj/item/clothing/suit/spaceblanket) && prob(70)))
+							M.ExtinguishMob()
+							visible_message("<span class='notice'>\The [user] has put out the fire on \the [target].</span>")
```

Comment:
I believe you're supposed to use present tense forms, e.g. "\The [user] puts out the fire"

Reply:
English isn't my first language. Will fix